### PR TITLE
Feature/channels

### PR DESCRIPTION
**Pull request by iPieter**

### CODE REVIEW

File: app/server.py [hunk 1]
```diff
@@ -181,6 +196,54 @@ def index():
...
 
 
+@app.route("/channels")
+@multi_auth.login_required
+def get_channels():
+    
+    channels = database.get_channels(g.user)
+    
+    for channel in channels:
+        channel['users'] = database.get_users_for_channel(channel['id'])
```
Comment (iPieter) [inline app/server.py:206]: I will fix this query

Comment (iPieter) [reply app/server.py:206]: This implementation takes 0.0060±0.0030 seconds (95% CI) for a user with 8 channels. I will investigate how an alternative query will perform.

Comment (iPieter) [reply app/server.py:206]: Is it ok if this stays as is, and in a later version the for-loop should be discarded altogether and replaced by client-side caching and on-demand fetching.

File: app/server.py [hunk 2]
```diff
@@ -47,6 +47,17 @@ def test_connect():
 def test_disconnect():
     print("Client disconnected")
 
+@socketio.on("join_rooms")
+def handle_join(token):
+    metadata = jws.loads(token)
+    print(metadata)
+    channels = database.get_channels(metadata['user'])
+    print(channels)
```
Comment (AntonDanneels) [inline app/server.py:55]: are these logs needed?

File: app/server.py [hunk 3]
```diff
@@ -47,6 +47,17 @@ def test_connect():
...
+    try:
+        for channel in channels:
+            join_room(channel['id'])    
+    except Exception as e:
```
Comment (AntonDanneels) [inline app/server.py:59]: why would this throw an exception?

File: app/server.py [hunk 4]
```diff
@@ -55,11 +66,11 @@ def handle_message(message):
         print(request.sid)
         # emit("msg", message, room=request.sid)
         msg_parsed = json.loads(message)
-        print(msg_parsed["sender"])
+        #(msg_parsed["sender"])
 
         metadata = jws.loads(msg_parsed["sender"])
-        print("parsed:")
-        print(metadata)
+        #print("parsed:")
```
Comment (AntonDanneels) [inline app/server.py:72]: remove these if not needed

File: app/server.py (hunk 2 above)
Comment (iPieter) [reply app/server.py]: Fixed in 15210fc

File: app/server.py (hunk 4 above)
Comment (iPieter) [reply app/server.py]: fixed in 15210fc

**Pull request merged**